library/psa_crypto.c

```diff
@@ -5011,7 +5102,7 @@ psa_status_t psa_key_derivation_set_capacity( psa_key_derivation_operation_t *op
     return( PSA_SUCCESS );
 }
 
-#if defined(MBEDTLS_MD_C)
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_HKDF)
```

Comment:
This guard only for psa_key_derivation_hkdf_read()
Guard with MBEDTLS_MD_C psa_key_derivation_tls12_prf_generate_next_block() and psa_key_derivation_tls12_prf_read()